packages/pull-signal/src/repositories/interop.repository.ts

```diff
@@ -56,13 +71,58 @@ export const interopRepository = (db: DB): IInteropRepository => {
            AND  agreement.eservice_id = eservice.eservice_id
            AND  UPPER(agreement.state) = UPPER($3)
            AND  purpose.consumer_id = agreement.consumer_id
-           AND  purpose.eservice_id = eservice.eservice_id 
+           AND  purpose.eservice_id = eservice.eservice_id
            AND  UPPER(purpose.purpose_state) = UPPER($4)`,
           [eserviceId, consumerId, agreementState, purposeState]
         );
       } catch (error: unknown) {
         throw genericError(`Error interopRepository::findBy ${error}`);
       }
+    },
+
+    async findAdminActsInDelegationBy(
+      eserviceId: string,
+      consumerId: string,
+      eserviceAllowedStates: string[],
+      purposeState: string,
+      agreementState: string,
+      delegationId: string
+    ): Promise<
+      {
+        eserviceId: string;
+        agreementId: string;
+        purposeId: string;
+      }[]
+    > {
+      const sqlConditionStates = eserviceAllowedStates
+        .map(
+          (eServiceState) => `UPPER(eservice.state) = UPPER('${eServiceState}')`
+        )
+        .join(" OR ");
+
+      try {
+        return await db.manyOrNone(
+          `SELECT
+                eservice.eservice_id eserviceId, agreement.agreement_id agreementId, purpose.purpose_id purposeId
+           FROM
+                ${eserviceTable} eservice, ${agreementTable} agreement, ${purposeTable} purpose
+           WHERE
+                eservice.eservice_id = $1
+           AND  eservice.enabled_signal_hub = true
+           AND  eservice.client_access_delegable = true
+           AND  (${sqlConditionStates})
+           AND  agreement.consumer_id = $2
+           AND  agreement.eservice_id = eservice.eservice_id
+           AND  UPPER(agreement.state) = UPPER($3)
+           AND  purpose.consumer_id = agreement.consumer_id
+           AND  purpose.eservice_id = eservice.eservice_id
+           AND  purpose.delegation_id = $5
+           AND  UPPER(purpose.purpose_state) = UPPER($4)`,
+          [eserviceId, consumerId, agreementState, purposeState, delegationId]
```

Comment:
just to make it clearer
```suggestion
           AND  purpose.delegation_id = $4
           AND  UPPER(purpose.purpose_state) = UPPER($5)`,
          [eserviceId, consumerId, agreementState, delegationId, purposeState]
```